File: Example/app/src/main/java/com/segmentify/segmentifyandroid/PreferencesManager.java
package com.segmentify.segmentifyandroid;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;
import com.securepreferences.SecurePreferences;

public class PreferencesManager {

    private static final String defaultPackageName = "mobi.appcent.defacto";
    private static final String emptyString = "";
    private final String preferenceName;
    private final SharedPreferences targetPreferences;
    private final Gson gson = new Gson();


    protected PreferencesManager(Context targetContext) {
        this(targetContext, defaultPackageName);
    }

    protected PreferencesManager(Context targetContext, String preferenceName) {
        this.preferenceName = preferenceName;
        String str = "test" + defaultPackageName;
        targetPreferences = new SecurePreferences(targetContext, str, "defacto_prefs.xml");
    }

    public void clearKey(String key) {
        if (targetPreferences.contains(key)) {
            putString(key, null);
        }
    }

    public boolean getBooleanValue(String key) {
        return targetPreferences.getBoolean(key, false);
    }

    public boolean getBooleanValue(String key, boolean defaultValue) {
        return targetPreferences.getBoolean(key, defaultValue);
    }

    public float getFloat(String key) {
        return targetPreferences.getFloat(key, 0);
    }

    public int getInt(String key) {
        return targetPreferences.getInt(key, 0);
    }

    public int getInt(String key, int defaultValue) {
        return targetPreferences.getInt(key, defaultValue);
    }

    public long getLong(String key) {
        return targetPreferences.getLong(key, 0);
    }

    public <T> T getObject(String key, Class<T> targetType) {
        return getObject(key, targetType, null);
    }

    public <T> T getObject(String key, Class<T> targetType, T defaultValue) {
        if (targetPreferences.contains(key)) {
            String preferenceTarget = targetPreferences.getString(key, "");
            if (!"".equals(preferenceTarget)) {
                return gson.fromJson(preferenceTarget, targetType);
            }
        }
        return defaultValue;
    }

    public String getPreferenceName() {
        return preferenceName;
    }

    public String getString(String key) {
        return getString(key, emptyString);
    }

    public String getString(String key, String defaultValue) {
        return targetPreferences.getString(key, defaultValue);
    }

    public boolean putBoolean(String key, boolean value) {
        SharedPreferences.Editor targetEditor = targetPreferences.edit();
        targetEditor.putBoolean(key, value);
        return targetEditor.commit();
    }

    public boolean putFloat(String key, float targetValue) {
        SharedPreferences.Editor targetEditor = targetPreferences.edit();
        return targetEditor.putFloat(key, targetValue).commit();
    }

    public boolean putInt(String key, int targetValue) {
        SharedPreferences.Editor targetEditor = targetPreferences.edit();
        return targetEditor.putInt(key, targetValue).commit();
    }

    public boolean putLong(String key, long targetValue) {
        SharedPreferences.Editor targetEditor = targetPreferences.edit();
        return targetEditor.putLong(key, targetValue).commit();
    }

    public void putObject(String key, Object targetObject) {
        putString(key, gson.toJson(targetObject));
    }

    public void putString(String[] keys, String[] values) {
        SharedPreferences.Editor targetEditor = targetPreferences.edit();
        int counter = 0;
        for (String key : keys) {
            targetEditor.putString(key, values[counter++]);
        }
        targetEditor.apply();
    }

    public void putString(String key, String value) {
        SharedPreferences.Editor targetEditor = targetPreferences.edit();
        targetEditor.putString(key, value);
        targetEditor.apply();
    }

}

File: segmentifyandroidsdk/src/main/java/com/segmentify/segmentifyandroidsdk/utils/PreferencesManager.kt
package com.segmentify.segmentifyandroidsdk.utils

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.Gson

abstract class PreferencesManager(targetContext: Context) {
    private val defaultPackageName = "AppcentPackageSettings"
    private val emptyString = ""
    private var preferenceName: String = ""
    private var targetPreferences: SharedPreferences
    private val gson = Gson()

    init {
        targetPreferences = targetContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE)
    }

    fun clearKey(key: String) {
        if (targetPreferences.contains(key)) {
            putString(key, null)
        }
    }

    fun getBooleanValue(key: String): Boolean {
        return targetPreferences.getBoolean(key, false)
    }

    fun getBooleanValue(key: String, defaultValue: Boolean): Boolean {
        return targetPreferences.getBoolean(key, defaultValue)
    }

    fun getFloat(key: String): Float {
        return targetPreferences.getFloat(key, 0f)
    }

    fun getInt(key: String): Int {
        return targetPreferences.getInt(key, 0)
    }

    fun getInt(key: String, defaultValue: Int): Int {
        return targetPreferences.getInt(key, defaultValue)
    }

    fun getLong(key: String): Long {
        return targetPreferences.getLong(key, 0)
    }

    fun getPreferenceName(): String {
        return preferenceName
    }

    fun getString(key: String): String? {
        return getString(key, emptyString)
    }

    fun getString(key: String, defaultValue: String): String? {
        return targetPreferences.getString(key, defaultValue)
    }

    fun putBoolean(key: String, value: Boolean): Boolean {
        val targetEditor = targetPreferences.edit()
        targetEditor.putBoolean(key, value)
        return targetEditor.commit()
    }

    fun putFloat(key: String, targetValue: Float): Boolean {
        val targetEditor = targetPreferences.edit()
        return targetEditor.putFloat(key, targetValue).commit()
    }

    fun putInt(key: String, targetValue: Int): Boolean {
        val targetEditor = targetPreferences.edit()
        return targetEditor.putInt(key, targetValue).commit()
    }

    fun putLong(key: String, targetValue: Long): Boolean {
        val targetEditor = targetPreferences.edit()
        return targetEditor.putLong(key, targetValue).commit()
    }

    fun putObject(key: String, targetObject: Any) {
        putString(key, gson.toJson(targetObject))
    }

    fun putString(keys: Array<String>, values: Array<String>) {
        val targetEditor = targetPreferences.edit()
        var counter = 0
        for (key in keys) {
            targetEditor.putString(key, values[counter++])
        }
        targetEditor.apply()
    }

    fun putString(key: String, value: String?) {
        val targetEditor = targetPreferences.edit()
        targetEditor.putString(key, value)
        targetEditor.apply()
    }


}